manifests/config.pp

```diff
@@ -8,17 +8,19 @@
 
   $etc_path = $ccs_software::etc_path
 
-  $etc_files = [
-    'logging.properties',
-    'ccsGlobal.properties',
-    'udp_ccs.properties',
-  ]
-
-  $epp_vars = {
-    'hostname' => $trusted['certname'],
+  ## Hash of templates and any arguments they take.
+  $etc_files = {
+    'logging.properties' => {},
+    'ccsGlobal.properties' => {
+      'global_properties' => lookup('ccs_software::global_properties', Array[String], 'unique', []),
```

Comment:
It isn't necessary to do a hiera lookup here as these parameters can be added to the `ccs_software` class and referred to directly.  That parameter approach is also more flexible as the module could be  configured both from puppet code and via hiera automatic binding.

Reply:
Does that do merging? I want the "unique" merge strategy.

Reply:
That policy can go into the hierarchy. E.g.: https://github.com/lsst-it/lsst-itconf/blob/master/hieradata/org/lsst.yaml#L2-L32

Reply:
Thanks for the feedback; see latest version.